### PR TITLE
Task trocar os logs do codex para yaml neodv 1304

### PR DESCRIPTION
**Pull request by henrique-lh**
## Description

With this pr:
* Improve log messages;
* Use YAML format instead of json (when possible);
* Refactor code (use early returns, improve methods);
* Apply RUFF to the project.

## Related issues

* Closed: https://linear.app/datarisk/issue/NEODV-1304/[task]-trocar-os-logs-do-codex-para-yaml

## How to test it

Test everything possible, but mainly the training and model modules

### CODE REVIEW

File: notebooks/Model.ipynb [hunk 1]
```diff
@@ -20,53 +20,53 @@
...
...zay5uZXQvdXNlci1hY3RpdmUiOnRydWUsImh0dHBzOi8vbmVvbWFyaWwuZGF0YXJpc2submV0L3JvbGUiOiJtYXN0ZXIiLCJpc3MiOiJodHRwczovL2Rldi1tazNvN2xhenhsZTMwaHdxLnVzLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw2NTY0Y2M0NTlkYzAzODhlNDVlMDQzZTciLCJhdWQiOlsiaHR0cHM6Ly9kZXYtbWszbzdsYXp4bGUzMGh3cS51cy5hdXRoMC5jb20vYXBpL3YyLyIsImh0dHBzOi8vZGV2LW1rM283bGF6eGxlMzBod3EudXMuYXV0aDAuY29tL3VzZXJpbmZvIl0sImlhdCI6MTcyOTE5NDQ0MiwiZXhwIjoxNzI5MjA1MjQyLCJzY29wZSI6Im9wZW5pZCBwcm9maWxlIGVtYWlsIGFkZHJlc3MgcGhvbmUgcmVhZDpjdXJyZW50X3VzZXIgdXBkYXRlOmN1cnJlbnRfdXNlcl9tZXRhZGF0YSBkZWxldGU6Y3VycmVudF91c2VyX21ldGFkYXRhIGNyZWF0ZTpjdXJyZW50X3VzZXJfbWV0YWRhdGEgY3JlYXRlOmN1cnJlbnRfdXNlcl9kZXZpY2VfY3JlZGVudGlhbHMgZGVsZXRlOmN1cnJlbnRfdXNlcl9kZXZpY2VfY3JlZGVudGlhbHMgdXBkYXRlOmN1cnJlbnRfdXNlcl9pZGVudGl0aWVzIG9mZmxpbmVfYWNjZXNzIiwiZ3R5IjoicGFzc3dvcmQiLCJhenAiOiJrckI2TWxHdmQ4R0FJQ1l3WE93SVptTkMwWTlUYVBBNyJ9.lA_inD8lMJ5CjDaH2X6j5kLl3G78wA4dg-GGdmOd_WcQexeLWGvsLtV0RXG683w7BcZRRkyrgQdpE0S6EECq0jEuPhgagkcruqEqz7-837l7gNt-FouVGnpp9yLdmgazJ07nTBmqA37XSUdpVUrUp4OAqTK9mnP2TnYHRmP2fJQNhHWAwWZkEbj-4y23dsZAkdmSuZKJn6TJaRJgfMEp00LprLS902IQezodsJup_65uMq_TnAN3mN45J4QQnT6l5j1rh9b06h2dMCajeLaHgBdwJvF3szDQrOIygkOGOUhtZgicTytl3vVEIGBRPmFURMVOa7UpVSqxeNR7zeWRHw\")"
```
Comment (64J0) [inline notebooks/Model.ipynb:57]: Is this a valid token? It must be removed from this document if this is the case @henrique-lh

File: notebooks/Training.ipynb [hunk 2]
```diff
@@ -43,20 +43,20 @@
...
...zay5uZXQvdXNlci1hY3RpdmUiOnRydWUsImh0dHBzOi8vbmVvbWFyaWwuZGF0YXJpc2submV0L3JvbGUiOiJtYXN0ZXIiLCJpc3MiOiJodHRwczovL2Rldi1tazNvN2xhenhsZTMwaHdxLnVzLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw2NTY0Y2M0NTlkYzAzODhlNDVlMDQzZTciLCJhdWQiOlsiaHR0cHM6Ly9kZXYtbWszbzdsYXp4bGUzMGh3cS51cy5hdXRoMC5jb20vYXBpL3YyLyIsImh0dHBzOi8vZGV2LW1rM283bGF6eGxlMzBod3EudXMuYXV0aDAuY29tL3VzZXJpbmZvIl0sImlhdCI6MTcyOTE5NDY3NywiZXhwIjoxNzI5MjA1NDc3LCJzY29wZSI6Im9wZW5pZCBwcm9maWxlIGVtYWlsIGFkZHJlc3MgcGhvbmUgcmVhZDpjdXJyZW50X3VzZXIgdXBkYXRlOmN1cnJlbnRfdXNlcl9tZXRhZGF0YSBkZWxldGU6Y3VycmVudF91c2VyX21ldGFkYXRhIGNyZWF0ZTpjdXJyZW50X3VzZXJfbWV0YWRhdGEgY3JlYXRlOmN1cnJlbnRfdXNlcl9kZXZpY2VfY3JlZGVudGlhbHMgZGVsZXRlOmN1cnJlbnRfdXNlcl9kZXZpY2VfY3JlZGVudGlhbHMgdXBkYXRlOmN1cnJlbnRfdXNlcl9pZGVudGl0aWVzIG9mZmxpbmVfYWNjZXNzIiwiZ3R5IjoicGFzc3dvcmQiLCJhenAiOiJrckI2TWxHdmQ4R0FJQ1l3WE93SVptTkMwWTlUYVBBNyJ9.d2OYwu7XVkBuyDAhRRnIabjsUNsUOxEFUafhGR2pGaxWbEN3pxBjrRdezoxyPB-Rj3U3mmphe25bu1kGDjWw-GfhN0v6y4-onRDNtKAgHidkl9DeJseNHe9zZE5z9fhjqU5M16isf5l5RrpWSBPLOEFJ8kZ-G5ynd6AfuBLIkwpwKOZgbbaRXp8TznpiSdGcJuHTzVDM3tRaDt50_U3vKNxg2UpFaALpUjAujw3PG0pwrttTgsETNuL8Lozd8aGHL2p60nTLatQOZamb8yR-DtljUfbM333IdsWn0tT5xCCwsec3tpSkXj-TZHv04AAn1QKrjF5cl7Z8AHZhig3v0g\")"
```
Comment (64J0) [inline notebooks/Training.ipynb:56]: The same here, it would be better to not even showing the shape of the token to be honest

File: src/neomaril_codex/__utils.py [hunk 3]
```diff
@@ -70,3 +71,15 @@ def refresh_token(login: str, password: str, base_url: str):
         return respose.json()["Token"]
     else:
         raise AuthenticationError(respose.text)
+
+
+def parse_json_to_yaml(json_dict) -> str:
+    """Parse a json file to yaml format
```
Comment (64J0) [inline src/neomaril_codex/__utils.py:77]: Does this function work with a json on memory too?

File: src/neomaril_codex/base.py [hunk 4]
```diff
@@ -218,23 +235,30 @@ def create_group(self, *, name: str, description: str) -> bool:
...
-            raise ServerError("Server Error")
-        else:
-            logger.error(response.text)
-            raise InputError("Bad Input. Client error")
+            return HTTPStatus.OK
+
+        formatted_msg = parse_json_to_yaml(response.json())
+
+        if response.status_code == 409:
+            logger.info(f"Something went wrong:\n {formatted_msg}")
```
Comment (64J0) [inline src/neomaril_codex/base.py:247]: Shouldn't this be a logger.error(...)?

File: src/neomaril_codex/base.py [hunk 5]
```diff
@@ -530,3 +567,7 @@ def download_result(
             raise ExecutionError("Execution failed")
         else:
             logger.info(f"Execution not ready. Status is {self.status}")
+
+    def execution_info(self) -> None:
+        """Show the execution data in a better format"""
+        print(parse_json_to_yaml(self.execution_data))
```
Comment (64J0) [inline src/neomaril_codex/base.py:573]: Why don't we use the logger instead of the print here?

File: src/neomaril_codex/logger_config.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import sys
+
+from loguru import logger
+
+
+def configure_logger(log_levels=["INFO", "ERROR"]):
+    logger.remove()
+
+    for level in log_levels:
+        logger.add(sys.stdout, level=level, format=f"{{time:MMMM D, YYYY}} | {level}: {{function}} {{message}}")
```
Comment (64J0) [inline src/neomaril_codex/logger_config.py:10]: Maybe we can have a "preprocess" step that changes the `log_levels` based on an environment variable. This way, it could be useful to still print the debug messages.

We can do it in another PR since this is out of the scope of this current issue.

**Review by 64J0 — COMMENTED** [review]
Added some comments

File: notebooks/Model.ipynb (hunk 1 above)
Comment (henrique-lh) [reply notebooks/Model.ipynb:57]: I tested locally, so it won't be accessible to users

File: src/neomaril_codex/__utils.py (hunk 3 above)
Comment (henrique-lh) [reply src/neomaril_codex/__utils.py:77]: If you try running the following code:

```python
import yaml

d = yaml.load("/path/to/file.json", Loader=yaml.CLoader)
print(d)
o = yaml.dump(d)
print(o)
```

it won’t display anything meaningful. This is because the file is in JSON format, not YAML. To fix this, you should load the file using the json module instead:

```python
import yaml
import json

with open("/home/henrique/datarisk/neomaril/samples/autoML/conf.json") as f:
    d = json.load(f)
print(d)
o = yaml.dump(d)
print(o)
```

File: src/neomaril_codex/base.py (hunk 4 above)
Comment (henrique-lh) [reply src/neomaril_codex/base.py]: https://github.com/datarisk-io/mlops-neomaril-codex/pull/59/commits/193f882f1dfb5bc3b33c0a490aff890e50b90d1f

File: src/neomaril_codex/base.py (hunk 5 above)
Comment (henrique-lh) [reply src/neomaril_codex/base.py:573]: I though about that, but before I create this method, in the codex sample the variable `execution_data` were accessed directly, printing a json, so I tried to keep this format. Change to a log would be better option?

File: src/neomaril_codex/logger_config.py (hunk 6 above)
Comment (henrique-lh) [reply src/neomaril_codex/logger_config.py:10]: Yeah, I was searching about it, but I couldn't thing on how to implement this. Maybe use a .env variable, but we would need to change the gitignore or try something else (I'm not sure)

File: src/neomaril_codex/__utils.py (hunk 3 above)
Comment (henrique-lh) [reply src/neomaril_codex/__utils.py:77]: Maybe change the name of the function to `parse_dict_to_yaml`

File: notebooks/Model.ipynb (hunk 1 above)
Comment (henrique-lh) [reply notebooks/Model.ipynb]: I decided to clear the outputs

https://github.com/datarisk-io/mlops-neomaril-codex/pull/59/commits/af0ee157f4f7189835746a16c0c39768199e8443

File: notebooks/Training.ipynb (hunk 2 above)
Comment (henrique-lh) [reply notebooks/Training.ipynb]: https://github.com/datarisk-io/mlops-neomaril-codex/pull/59/commits/af0ee157f4f7189835746a16c0c39768199e8443

File: src/neomaril_codex/__utils.py (hunk 3 above)
Comment (64J0) [reply src/neomaril_codex/__utils.py:77]: I meant the documentation, it states that this function works with a json file instead of a json in memory.

File: src/neomaril_codex/base.py (hunk 5 above)
Comment (64J0) [reply src/neomaril_codex/base.py:573]: Yes, I think it makes sense to have a more consistent code regarding this information presentation level.

Comment (henrique-lh) [reply src/neomaril_codex/base.py]: https://github.com/datarisk-io/mlops-neomaril-codex/pull/59/commits/a3a361ad52668aca74056c5581a3c9c0d3891981

File: src/neomaril_codex/__utils.py (hunk 3 above)
Comment (henrique-lh) [reply src/neomaril_codex/__utils.py]: https://github.com/datarisk-io/mlops-neomaril-codex/pull/59/commits/3e5754974e16607c13f3f2765babda81d6c064db

**Review by 64J0 — APPROVED** [review]
